services/requester/cadence/call.cdc

```diff
@@ -1,15 +1,16 @@
 import EVM
 
 access(all)
-fun main(data: [UInt8], contractAddress: [UInt8; 20]): [UInt8] {
+fun main(hexEncodedData: String, contractAddress: [UInt8; 20]): [UInt8] {
```

Comment:
Maybe the contract address could also be hex-encoded to reduce the size of the transaction more.

In https://github.com/onflow/cadence/pull/3028, variable-sized functions got a `toConstantSized` function, so something like `hexEncodedContractAddress: String` and `hexEncodedContractAddress.decodeHex().toConstantSized<[UInt8; 20]>()` should work.

Reply:
Oh, that's nice :ok_hand: I wasn't aware this was introduced already.

Reply:
can we update this with other scripts as well then (get balance). This is much nicer.

Reply:
can we do same change for address on get balance script as well as get nonce

Reply:
Updated in https://github.com/onflow/flow-evm-gateway/pull/124/commits/9acbc3100f7d80be75e1e1e1b48f9c7630aabb54

Reply:
Nice!